compiler/rustc_parse/src/parser/generics.rs

```diff
@@ -284,7 +284,7 @@ impl<'a> Parser<'a> {
             }))
         // FIXME: Decide what should be used here, `=` or `==`.
         // FIXME: We are just dropping the binders in lifetime_defs on the floor here.
-        } else if self.eat(&token::Eq) || self.eat(&token::EqEq) {
```

Comment:
This would be better replaced with a diagnostic suggesting the user uses `=` instead.